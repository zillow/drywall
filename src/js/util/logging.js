export const log = (message, level = 'warn') => {
    if (typeof process !== 'undefined' && process.env.NODE_ENV !== 'production') {
        // eslint-disable-next-line no-console
        console[level](message);
    }
};

export const logDeprecation = message => {
    log(`[Deprecation] ${message}`);
};
