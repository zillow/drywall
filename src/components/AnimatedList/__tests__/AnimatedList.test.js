import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeProvider, css, isStyledComponent } from 'styled-components';
import { AnimatedList } from '../../../index';
import { createTestTheme } from '../../../test/util';

describe('<AnimatedList>', () => {
    it('can render without theme', () => {
        const testRenderer = TestRenderer.create(
            <AnimatedList>{['one', 'two', 'three']}</AnimatedList>
        );
        expect(testRenderer.toJSON()).toMatchSnapshot();
    });

    describe('without Object.entries', () => {
        const { entries } = global.Object;
        beforeEach(() => {
            delete global.Object.entries;
        });
        afterEach(() => {
            global.Object.entries = entries;
        });
        it('renders a static list when react-spring is not supported', () => {
            const testRenderer = TestRenderer.create(
                <AnimatedList>{['one', 'two', 'three']}</AnimatedList>
            );
            expect(testRenderer.toJSON()).toMatchSnapshot();
        });
    });

    it('can render with theme', () => {
        const theme = createTestTheme({
            AnimatedList: css`
                list-style: none;
            `,
        });
        const testRenderer = TestRenderer.create(
            <ThemeProvider theme={theme}>
                <AnimatedList>{['one', 'two', 'three']}</AnimatedList>
            </ThemeProvider>
        );
        const tree = testRenderer.toJSON();
        expect(tree).toHaveStyleRule('list-style', 'none');
        expect(tree).toMatchSnapshot();
    });

    it('is a styled-component', () => {
        expect(isStyledComponent(AnimatedList)).toBe(true);
    });
});
