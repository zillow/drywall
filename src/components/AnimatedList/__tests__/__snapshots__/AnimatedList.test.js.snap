// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnimatedList> can render with theme 1`] = `
.c0 {
  list-style: none;
}

<ul
  className="c0"
>
  <li
    className="AnimatedList__Item-sdziqv-0 "
    style={
      Object {
        "height": "auto",
        "opacity": 1,
        "position": "absolute",
        "visibility": "hidden",
      }
    }
  >
    one
  </li>
  <li
    className="AnimatedList__Item-sdziqv-0 "
    style={
      Object {
        "height": "auto",
        "opacity": 1,
        "position": "absolute",
        "visibility": "hidden",
      }
    }
  >
    two
  </li>
  <li
    className="AnimatedList__Item-sdziqv-0 "
    style={
      Object {
        "height": "auto",
        "opacity": 1,
        "position": "absolute",
        "visibility": "hidden",
      }
    }
  >
    three
  </li>
</ul>
`;

exports[`<AnimatedList> can render without theme 1`] = `
<ul
  className="AnimatedList-sdziqv-1 "
>
  <li
    className="AnimatedList__Item-sdziqv-0 "
    style={
      Object {
        "height": "auto",
        "opacity": 1,
        "position": "absolute",
        "visibility": "hidden",
      }
    }
  >
    one
  </li>
  <li
    className="AnimatedList__Item-sdziqv-0 "
    style={
      Object {
        "height": "auto",
        "opacity": 1,
        "position": "absolute",
        "visibility": "hidden",
      }
    }
  >
    two
  </li>
  <li
    className="AnimatedList__Item-sdziqv-0 "
    style={
      Object {
        "height": "auto",
        "opacity": 1,
        "position": "absolute",
        "visibility": "hidden",
      }
    }
  >
    three
  </li>
</ul>
`;

exports[`<AnimatedList> without Object.entries renders a static list when react-spring is not supported 1`] = `
<ul
  className="AnimatedList-sdziqv-1 "
>
  <li
    className="AnimatedList__Item-sdziqv-0 "
  >
    one
  </li>
  <li
    className="AnimatedList__Item-sdziqv-0 "
  >
    two
  </li>
  <li
    className="AnimatedList__Item-sdziqv-0 "
  >
    three
  </li>
</ul>
`;
